packages/flutter_tools/test/host_cross_arch.shard/cache_test.dart

```diff
@@ -31,6 +31,12 @@ HostPlatform _identifyMacBinaryArch(String path) {
   final ProcessResult result = processManager.runSync(
     <String>['file', _dartBinary.path],
   );
+  expect(
+      result,
+      ProcessResultMatcher(
+        0,
+        stdoutSubstring: '${_dartBinary.path}: Mach-O 64-bit executable',
+      ));
```

Comment:
An additional example usage of `ProcessResultMatcher` that also matches on the `exitCode` and a substring to find in `stdout` for a given `ProcessResult`